drivers/sensor: add support to LSM6DSV16X IMU sensor

avisconti opened the pull request:
The LSM6DSV16X is a system-in-package featuring a 3-axis digital accelerometer and a 3-axis digital gyroscope for industrial and IoT solutions. The LSM6DSV16X embeds advanced dedicated features such as a finite state machine (FSM) for configurable motion tracking and a machine learning core (MLC) for context awareness.

https://www.st.com/en/mems-and-sensors/lsm6dsv16x.html

This driver is based on stmemsc HAL i/f v2.02
It follows the inclusive language strategy adopted in #56800

avisconti committed:
drivers/sensor: add support to LSM6DSV16X IMU sensor

The LSM6DSV16X is a system-in-package featuring a 3-axis digital
accelerometer and a 3-axis digital gyroscope for industrial and IoT
solutions. The LSM6DSV16X embeds advanced dedicated features such as
a finite state machine (FSM) for configurable motion tracking and a
machine learning core (MLC) for context awareness.

https://www.st.com/en/mems-and-sensors/lsm6dsv16x.html

This driver is based on stmemsc HAL i/f v2.02

Signed-off-by: Armando Visconti <armando.visconti@st.com>

MaureenHelm commented on drivers/sensor/lsm6dsv16x/lsm6dsv16x.h:
```diff
@@ -0,0 +1,141 @@
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/types.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
```
You're including this header twice, remove this one

avisconti replied:
thx

MaureenHelm commented on drivers/sensor/lsm6dsv16x/lsm6dsv16x_trigger.c:
```diff
@@ -0,0 +1,332 @@
+
+	/* set interrupt (TEMP DRDY interrupt is only on INT2) */
+	if (cfg->drdy_pin == 1)
+		return -EIO;
```
wrap in braces

avisconti replied:
ack

MaureenHelm commented on tests/drivers/build_all/sensor/i2c.dtsi:
```diff
@@ -728,3 +728,9 @@ test_i2c_lsm6dso16is: lsm6dso16is@6f {
 	irq-gpios = <&test_gpio 0 0>;
 };
+
+test_i2c_lsm6dsv16x: lsm6dsv16x@6b {
```
Update the address

avisconti replied:
ack

MaureenHelm commented on tests/drivers/build_all/sensor/spi.dtsi:
```diff
@@ -388,3 +388,10 @@ test_spi_lsm6dso16is: lsm6dso16is@2f {
 	irq-gpios = <&test_gpio 0 0>;
 };
+
+test_spi_lsm6dsv16x: lsm6dsv16x@2f {
```
update the address

avisconti replied:
ack, thx

MaureenHelm reviewed (changes requested):
Some small nits, but otherwise LGTM

avisconti committed:
drivers/sensor: lsm6dso16is: fix coding style issues

1. spi.h is included twice. Remove one of the two "#include"
   declarations.

2. The 'if' clause requires braces even for one line body.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

teburd reviewed (approved):
Looks great
